File: 351002/Lashchenka/discussion/src/main/java/by/bsuir/distcomp/discussion/service/ModerationService.java
package by.bsuir.distcomp.discussion.service;

import by.bsuir.distcomp.discussion.model.ReactionState;
import org.springframework.stereotype.Service;

import java.util.Locale;
import java.util.Set;

@Service
public class ModerationService {

    private static final Set<String> STOP_WORDS = Set.of(
            "spam", "scam", "hate", "сука", "дурак", "мат", "запрещено"
    );

    public ReactionState moderate(String content) {
        if (content == null || content.isBlank()) {
            return ReactionState.DELCINE;
        }
        String lower = content.toLowerCase(Locale.ROOT);
        for (String w : STOP_WORDS) {
            if (lower.contains(w.toLowerCase(Locale.ROOT))) {
                return ReactionState.DELCINE;
            }
        }
        return ReactionState.APPROVE;
    }
}

File: 351003/Soika/entities-app/discussion/src/main/kotlin/com/example/entitiesapp/service/ModerationService.kt
package com.example.entitiesapp.service

import com.example.entitiesapp.dto.CommentState
import org.springframework.stereotype.Service

@Service
class ModerationService {
    private val forbiddenWords = listOf("спам", "реклама", "badword")

    fun moderate(content: String): CommentState {
        return if (forbiddenWords.any { content.contains(it, ignoreCase = true) }) {
            CommentState.DECLINE
        } else {
            CommentState.APPROVE
        }
    }
}